File: gcc/cpp/Gcd/gcd.cpp
#include <fstream>
#include <iostream>
//gcd file

int gcd(int num1,int num2){
	if(num2==0) return num1;
	else return gcd(num2,num1%num2);
}

int main(){
	std::ifstream infile;
	std::ofstream exitfile;
	std::ifstream golden;
	std::ofstream result;
	result.open("resultGcd.txt");
	golden.open("goldenGcd.txt");
	exitfile.open("exitGcd.txt");
	infile.open("inputGcd.txt");
	float a,b;
	if(!infile){
		exitfile<<"ERROR from inputGcd.txt"<<std::endl;
	}
	else {
		while(infile>>a>>b){
			exitfile<<gcd(a,b)<<std::endl;
		}
		exitfile.close();
		std::ifstream exitfile;
		exitfile.open("exitGcd.txt");
		float c,d;
		int number = 0;
		int procent = 100;
		while(golden>>c && exitfile>>d){
			number++;
			if(c == d) result<<c<<"\t"<<d<<"\t test "<<number<<" passed!"<<std::endl;
			else{
				result<<c<<"\t"<<d<<"\ttest "<<number<<" failed!"<<std::endl;
				procent -=10;
			}
		}
		if(procent>=50) result<<std::endl<<"Tests passed Succesfully:"<<std::endl<<"Tests Result:  "<<procent<<"%";
	}
	golden.close();
	result.close();
	exitfile.close();
	infile.close();
	std::cout<<"run gcd"<<std::endl;
	return 0;
}

File: final_version/Gcd/Js/gcd.js
let fs = require('fs');
const gcd = (num1, num2) => {
	if(num2 === 0) return num1;
	return gcd(num2, num1 % num2);
}

let input = fs.readFileSync('input.txt','utf-8');
let data = input.split('\n').join(' ').split(' ')
data.pop();
let length = data.length;
var exit1 = '';
for(let i=0; i<length; i+=2){
	exit1+=`${gcd(data[i],data[i+1])}\n`;
}
fs.writeFile('exit.txt',exit1,()=>{});


const test = () => {
	let golden = fs.readFileSync('golden.txt','utf-8').split('\n');
	let exit = exit1.split('\n');
	let str = '';
	let procent = 100;
	let length = golden.length-1;
	for(let i = 0; i<length; i++){
		if(golden[i] === exit[i])
			str += `${golden[i]} = ${exit[i]} test ${i+1} passed \t <3\n`;
		else{
			str += `${golden[i]} != ${exit[i]} test ${i+1} failed \t :(\n`
			procent -=10}
	}

	str +=`\n\nTesting Result:  ${procent}%`

	let result = fs.writeFile('result.txt', str , ()=>{});
}

test();
